grid.v3/codec/registry.go

```diff
@@ -1,61 +1,90 @@
 package codec
 
 import (
+	"errors"
 	"reflect"
 	"sync"
+
+	"github.com/gogo/protobuf/proto"
 )
 
-var registry *codecRegistry = &codecRegistry{reg: map[string]Codec{}} //shared registry
+var (
+	ErrNonProtoMessage         = errors.New("codec: non proto message")
```

Comment:
You're right about it being strange. Actually it's probably best that it does not even exist. It's used when a type assertion fails, but that assertion can only fail if the registry has a fatal bug, so I'll turn it into a panic, since it's supposed to be an invariant that you can only register protobuf messages, and the `Register` function actually checks that.